base/mpfr.jl

```diff
@@ -338,8 +338,18 @@ Float32(x::BigFloat, r::MPFRRoundingMode=ROUNDING_MODE[]) =
     _cpynansgn(ccall((:mpfr_get_flt,:libmpfr), Float32, (Ref{BigFloat}, MPFRRoundingMode), x, r), x)
 Float32(x::BigFloat, r::RoundingMode) = Float32(x, convert(MPFRRoundingMode, r))
 
-# TODO: avoid double rounding
-Float16(x::BigFloat) = Float16(Float64(x))
+function Float16(x::BigFloat) :: Float16
+    res = Float32(x)
+    resi = reinterpret(UInt32, res)
+    if (resi&0x7fffffff) < 0x38800000
```

Comment:
Maybe this starts to need some comments now. It's a bit of magic number soup now.

Reply:
good point. that's just if `Float16(res)` is subnormal.

Reply:
And this is more efficient than `issubnormal(Float16(res))`?

Reply:
yeah. this is only 2 instructions while converting to float16 is relatively expensive.